scalikejdbc-core/src/main/scala/scalikejdbc/StatementExecutor.scala

```diff
@@ -101,17 +101,17 @@ case class StatementExecutor(
       case p: org.joda.time.LocalDate => underlying.setDate(i, p.toDate.toSqlDate)
       case p: org.joda.time.LocalTime => underlying.setTime(i, p.toSqlTime)
       case p: java.time.ZonedDateTime =>
-        underlying.setTimestamp(i, java.util.Date.from(p.toInstant).toSqlTimestamp)
+        underlying.setTimestamp(i, java.sql.Timestamp.from(p.toInstant))
       case p: java.time.OffsetDateTime =>
-        underlying.setTimestamp(i, java.util.Date.from(p.toInstant).toSqlTimestamp)
+        underlying.setTimestamp(i, java.sql.Timestamp.from(p.toInstant))
       case p: java.time.Instant =>
-        underlying.setTimestamp(i, new java.util.Date(p.toEpochMilli).toSqlTimestamp)
+        underlying.setTimestamp(i, java.sql.Timestamp.from(p))
       case p: java.time.LocalDateTime =>
-        underlying.setTimestamp(i, org.joda.time.LocalDateTime.parse(p.toString).toDate.toSqlTimestamp)
+        underlying.setTimestamp(i, java.sql.Timestamp.valueOf(p))
       case p: java.time.LocalDate =>
-        underlying.setDate(i, org.joda.time.LocalDate.parse(p.toString).toDate.toSqlDate)
+        underlying.setDate(i, java.sql.Date.valueOf(p))
       case p: java.time.LocalTime =>
-        underlying.setTime(i, org.joda.time.LocalTime.parse(p.toString).toSqlTime)
+        underlying.setObject(i, p)
```

Comment:
Just in case, passing `java.sql.Time` value may be still safer. I am not sure all JDBC drivers handle `java.time.*` objects inside `PreparedStatement#setObject`.

Reply:
http://download.oracle.com/otndocs/jcp/jdbc-4_2-mrel2-spec/index.html

<img width="670" alt="spec-1" src="https://user-images.githubusercontent.com/389787/33805099-3302f58a-ddf6-11e7-87a4-40c2efd709ab.png">

<img width="611" alt="spec-2" src="https://user-images.githubusercontent.com/389787/33805114-68580e96-ddf6-11e7-986a-6ae3f8f0a7f3.png">

Reply:
How about https://github.com/xuwei-k/scalikejdbc/commit/17f8052c91de2b365516525ef9cc52b1bf1eee62 ?

Reply:
LGTM. Thank you for your effort 🙇 

```scala
scala> val p = java.time.LocalTime.now
p: java.time.LocalTime = 21:36:27.791

scala> val LocalDateEpoch = java.time.LocalDate.ofEpochDay(0)
LocalDateEpoch: java.time.LocalDate = 1970-01-01

scala> val offset = java.time.OffsetDateTime.now().getOffset()
offset: java.time.ZoneOffset = +09:00

scala> val i = p.atDate(LocalDateEpoch).toInstant(offset).toEpochMilli()
i: Long = 45387791

scala> val t = new java.sql.Time(i)
t: java.sql.Time = 21:36:27

scala> t.getTime
res25: Long = 45387791
```